### DISCUSSION TITLE
After latest update - Installation fails

### DISCUSSION THREAD

**Discussion by gusinghbal** [jasonacox/Powerwall-Dashboard#518] [Q&A]
Hi @jasonacox and team,
First of all, excellent dashboard , great work, Thank you.
I had the dashboard working on my RPi without issues for a couple of weeks. I then had some issues with my Pi and I had to re-install from fresh. I have all the notes from previous install and the instructions here off course. I am following the same instructions, but the setup.sh is not running anymore. I get to the following and then nothing further.

dietpi@DietPi:~/Powerwall-Dashboard$ ./setup.sh
Powerwall Dashboard (v4.5.2) - SETUP
"-----------------------------------------"

dietpi@DietPi:~/Powerwall-Dashboard$

Has anything changed in the latest update or am I doing something wrong?

Any help will be awesome.

Thanks.

**Comment by gusinghbal** [comment]
I believe, I may have found the issue and a solution.
Once I installed PIP , setup worked without issues.

**Reply by jasonacox** [reply]
Thanks @gusinghbal - This is puzzling. We don't use/require pip in the setup.sh script.  I could add a check for it if that was what was needed, but I'm afraid it may break on systems that don't have or need pip.

I'll try to schedule a RPi reload to simulate what you did to see what I'm missing. Thanks for reporting this.

**Reply by gusinghbal** [reply]
I just confirmed again on a fresh install. The setup wouldn't move until I install PIP and it installs without any issues. I am 100% certain that I installed it just a few weeks ago without PIP. Is it possible, anything changed in the past week or so?

**Reply by jasonacox** [reply]
The latest change does introduce logic to detect if you have python3 or python installed and uses that to find your geo-coordinates. I wonder if you installing pip also installs python so that is why it works?
